src/Database/DatabaseConfig.php

```diff
@@ -158,4 +201,27 @@ public function manager(): Contracts\TenantDatabaseManager
 
         return $databaseManager;
     }
+
+    /** Get the TenantDatabaseManager for this tenant's connection. */
+    public function hostManager(): Contracts\TenantDatabaseManager
+    {
+        $this->purgeHostConnection();
```

Comment:
Add comments explaining what is being done here, at all of the important/unclear lines.

In general, a method used for *fetching something* (here that is fetching the host manager), shouldn't have many side effects ideally.

Reply:
Also add a `// todo` with a note to rename this method, I think the name isn't ideal but can't come up with a better one right now.

Reply:
Added the todo and possible comments. TBH, I don't see unclear code here, almost every line is pretty clear what it is doing. 

> In general, a method used for fetching something (here that is fetching the host manager), shouldn't have many side effects, ideally.

What do you mean by "shouldn't have many side effects"? What side effects this method has? This method actually creates and returns the host manager, not just **fetching the host manager**.

Reply:
> What side effects this method has?

What's the `$this->purgeHostConnection()` call for? Is that for purging the connection used specifically by the returned manager?

At first look it seems like it's making a change to some general connection, but if it's specifically for the connection used by the returned hostManager then that's fine. I'd just add some comments to make it a bit more clear at first look.

> This method actually creates and returns the host manager, not just fetching the host manager.

It doesn't matter *how* it does it, my point is that it's a method for *getting something*. It creates the manager and returns it.

That's what I mean by fetching. I mean that it's not an action that does some side effects, instead it's a method for fetching some object. But as I mentioned above it's possible that it doesn't have any general side effects, so it might be fine.

So the only change I'd make here would be adding those comments. The code may be clear when you take a deep look at it, but we generally aim for code being immediately obvious upon seeing it. 

This method has a lot of steps for a method that just creates and returns an object, so the steps which:
1. Purge the host connection
2. Manipulate config
3. Access the driver

Should have simple comments explaining what we're doing and why we're doing it.

Reply:
> What's the $this->purgeHostConnection() call for? Is that for purging the connection used specifically by the returned manager?

Yes, it is. I thought the method name `purgeHostConnection` is pretty much clear that we are purging the host connection, not any connection.

Reply:
Resolving this because comments and todos were added.